android/capacitor/src/main/java/com/getcapacitor/CapConfig.java

```diff
@@ -141,7 +465,163 @@ private String getConfigKey(String key) {
     }
+
+    /**
+     * Get a JSON object value from the given JSON object.
+     *
+     * @param jsonObject A JSON object to search
+     * @param key A key to fetch from the JSON object
+     * @return The value from the config, if exists. Null if not
+     */
+    private JSONObject getObject(JSONObject jsonObject, String key) {
+        String k = getDeepestKey(key);
+        try {
+            JSONObject o = getDeepestObject(jsonObject, key);
+
+            return o.getJSONObject(k);
+        } catch (Exception ignore) {
+            // value was not found
+        }
+        return null;
+    }
+
+    /**
+     * Given a JSON key path, gets the deepest key.
+     *
+     * @param key The key path
+     * @return The deepest key
+     */
+    private String getDeepestKey(String key) {
+        String[] parts = key.split("\\.");
+        if (parts.length > 0) {
+            return parts[parts.length - 1];
+        }
+
+        return null;
+    }
+
+    /**
+     * Given a JSON object and key path, gets the deepest object in the path.
+     *
+     * @param jsonObject A JSON object
+     * @param key The key path to follow
+     * @return The deepest object along the key path
+     * @throws JSONException Thrown if any JSON errors
+     */
+    private JSONObject getDeepestObject(JSONObject jsonObject, String key) throws JSONException {
+        String[] parts = key.split("\\.");
+        JSONObject o = jsonObject;
+
+        // Search until the second to last part of the key
+        for (int i = 0; i < parts.length - 1; i++) {
+            String k = parts[i];
+            o = o.getJSONObject(k);
+        }
+
+        return o;
+    }
+
+    /**
+     * Builds a Capacitor Configuration in code
+     */
+    public static class Builder {
+
+        // Server Config Values
+        private boolean html5mode = true;
+        private String serverUrl;
+        private String hostname = "localhost";
+        private String androidScheme = CAPACITOR_HTTP_SCHEME;
+        private String[] allowNavigation;
+
+        // Android Config Values
+        private String overriddenUserAgentString;
+        private String appendedUserAgentString;
+        private String backgroundColor;
+        private boolean allowMixedContent = false;
+        private boolean captureInput = false;
+        private Boolean webContentsDebuggingEnabled;
+        private boolean hideLogs = false;
+
+        // Plugins Config Object
+        private JSONObject pluginConfigurations = new JSONObject();
+
+        /**
+         * Builds a Capacitor Config from the builder.
+         *
+         * @return A new Capacitor Config
+         */
+        public CapConfig create() {
+            return new CapConfig(this);
+        }
+
+        public Builder pluginConfigurations(JSONObject pluginConfigurations) {
+            this.pluginConfigurations = pluginConfigurations;
+            return this;
+        }
+
+        public Builder html5mode(boolean html5mode) {
```

Comment:
I think these should follow the setter pattern: `setHTML5Mode()`, `setServerURL`(), etc.

Reply:
They dont need to, but it seems Android does that so I'll copy their style